featurecontrol/featurecontrol.go

```diff
@@ -22,24 +22,31 @@ import (
 )
 
 const (
-	fcReceiverNameInMetrics = "receiver-name-in-metrics"
+	fcReceiverNameInMetrics    = "receiver-name-in-metrics"
+	fcDisabledNewLabelMatchers = "disable-new-label-matchers"
```

Comment:
```suggestion
	fcUTF8MatcherParsing = "utf8-matchers-parsing"
	fcStrictMatcherParsing = "strict-matchers-parsing"
```

Reply:
One is more restrictive than the other, so that's the name I came up with.

The feature flags have wording around them in terms of enablement - as such, I think enabling and disabling sounds counterintuitive.

Reply:
What's the difference between `utf8-matchers-parsing` and `strict-matchers-parsing`?

Reply:
`utf8-matchers-parsing` is new parser and `strict-matchers-parsing` is the old parser - not sure if this is what you're asking though so feel free to clarify.

Reply:
Hmm. But the old parser is less strict than the new the parser? In either case, I was thinking of just having one feature flag rather than two, as with two we need to add code to detect when both are enabled.

Reply:
Perhaps we're talking about different restrictions - the new parser is more strict in terms of how it accepts inputs but less restrictive in terms of what kind of matchers it accepts (we accept any UTF-8 valid character as label name after all).

Reply:
>In either case, I was thinking of just having one feature flag that disables the new parser as that's what we discussed together with Simon in the AM Working Group. If we have two feature flags I don't understand what we're supposed to do when either both feature flags are enabled or no feature flags are enabled?

For completeness we basically have 3 modes:

- just old parser
- just new parser
- both new and old parser

In my eyes, 2 new flags + a default doesn't seem too out of this world.

Reply:
OK! I think in such case we should find another name instead of `strict-matchers-parsing` because we are both extending and restricting the grammar with the new parser. Perhaps `stable-matchers-parsing`?

Yes, but in the Working Group I thought we agreed on just two modes. How come we're changing it now?

Reply:
OK! I think in such case we should find another name instead of strict-matchers-parsing because we are both extending and restricting the grammar with the new parser. Perhaps stable-matchers-parsing?

>F or completeness we basically have 3 modes:

Yes, but in the Working Group I thought we agreed on just two modes. How come we're changing it?

Reply:
@simonpasquier do you have any concerns with the above approach?

My thinking of having 3 modes (default + 2 flags) is that:

1.  Early adopters and innovators can run with the flag of "only new parser" enabled (this future proofing them for when/if we make the new parser permanent)
2. The flag of "only old parser" it's there in case we have any sort of bugs on the new parser where we only want to enable old parsing to make sure users are not "stuck".
3. The default mode of having both parsers working together to ensure most people are able to upgrade as seamlessly as possible w/o any sort of breaking change.

The approach I'd take is that as part of the changelog, we recommend people to run with the flag of `--new-parser-enabled` if they have the time to fix things between upgrades as that would guarantee they are future proof.

Reply:
I pushed a commit for the two-option flag (+ default) so we can see how it looks.